docs/develop/dialog-api-in-office-add-ins.md

```diff
@@ -293,11 +330,48 @@ function onRegisterMessageComplete(asyncResult) {
 
+### Cross-domain messaging to the dialog runtime
+
+The dialog may be navigated away from the add-in's domain or the parent JavaScript runtime (either in a task pane or a UI-less runtime that hosts a function file) may be navigated away from the add-in's domain after the dialog is opened. If either of these things has happened, then a call of `messageChild` will fail unless your code specifies the domain of the dialog runtime. You do this by adding a [DialogMessageOptions](/javascript/api/office/office.dialogmessageoptions) parameter to the call of `messageChild`. This object has a `targetOrigin` property that specifies the domain to which the message should be sent. If the parameter is not used, Office assumes that the target is the same domain that the parent runtime is currently hosting. 
+
+> [!NOTE]
+> Using `messageChild` to send a cross-domain message requires the [Dialog Origin 1.1 requirement set](../reference/requirement-sets/dialog-origin-requirement-sets.md).
+
+The following is an example of using `messageChild` to send a cross-domain message.
+
+```js
+dialog.messageChild(messageToDialog, { targetOrigin: "https://target.domain.com" });
+```
+
+If the message doesn't include sensitive data, you can set the `targetOrigin` to "\*" which allows it to be *sent* to any domain. The following is an example.
+
+```js
+dialog.messageChild(messageToDialog, { targetOrigin: "*" });
+```
+
+The JavaScript runtime that is hosting the dialog cannot access the `<AppDomains>` section of the manifest to ensure that the domain *from which the message comes* is trusted, so an alternative has been provided. The object that is passed to the `DialogParentMessageReceived` handler contains the domain that is currently hosted in the dialog as its `origin` property. Your code should use the handler to verify that the message is coming from a trusted domain. The following is an example.
+
+```javascript
+function onMessageFromParent(arg) {
+    if (arg.origin === "https://some.trusted.domain.com") {
+        // process message
+    } else {
+        dialog.close();
+        showNotification("Messages from " + arg.origin + " are not accepted.");
+    }
+}
+```
+
+For example, your code could use the [Office.onReady or Office.initialize methods](initialize-add-in.md) to store an array of trusted domains in a global variable. The `arg.origin` property could then be checked against that list in the handler.
+
+> [!TIP]
+> The `DialogMessageOptions` parameter was added to the `messageChild` method as a required parameter in mid-2021. Older add-ins that send a cross-domain message with the method no longer work until they are updated to use the new parameter. In the meantime, *on Office for Windows only*, users and system administrators can enable those add-ins to continue working by specifying the trusted domain(s) with a registry setting. See the **Tip** under [Cross-domain messaging to the host runtime](#cross-domain-messaging-to-the-host-runtime) for details.
```

Comment:
Maybe just put the tip in both places? I personally would be annoyed to have to scroll up to re-read a tip (or read it for the first time if I somehow skipped it earlier). Also because this is a breaking change, I think it warrants something other than tip